assets/src/components/searchForm.tsx

```diff
@@ -54,7 +54,82 @@ type SearchFormProps = SearchFormEventProps & {
   /**
    * Callback to run when {@link filters} should be updated.
    */
-  onFiltersChanged: (name: SearchProperty, currentValue: boolean) => void
+  onFiltersChanged: (searchFilterState: SearchFiltersState) => void
+}
+
+const allFiltersOn: SearchFiltersState = {
+  vehicle: true,
+  operator: true,
+  run: true,
+  location: true,
+}
+const allFiltersOff: SearchFiltersState = {
+  vehicle: false,
+  operator: false,
+  run: false,
+  location: false,
+}
+
+const Filters = ({
+  filters,
+  onFiltersChanged,
+}: {
+  filters: SearchFiltersState
+  onFiltersChanged: (searchFilterState: SearchFiltersState) => void
+}) => {
+  const countActiveFilters = Object.values(filters).filter(
+    (isActive) => isActive
+  ).length
+  const [hasAManuallyActivatedFilter, setHasAManuallyActivatedFilter] =
+    useState(
+      countActiveFilters > 0 && countActiveFilters < Object.keys(filters).length
+    )
```

Comment:
**nitpick (non-blocking):** not sure if this matters, but one thing I'm noticing is that because this is tied to the component state and not the global search page state, if you toggle all of the filters on, go to another mode, and then come back to search it will show the version with all of the toggles inactive again. Maybe a total non-issue but I thought I'd point it out.

Reply:
Ooh interesting! I'll flag that for design review. I'm hoping this is OK as-is, especially knowing there are planned changes to this filter component which may or may not remove the need for this logic.